pkg/providers/aws/cluster_network_provider.go

```diff
@@ -0,0 +1,87 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/integr8ly/cloud-resource-operator/pkg/resources"
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	errorUtil "github.com/pkg/errors"
+)
+
+//todo we need to ensure we test that if `IsEnabled` is true, it remains true for consecutive reconciles
+
+type NetworkManager interface {
+	CreateNetwork() error
+	DeleteNetwork() error
+	IsEnabled(context.Context, client.Client, ec2iface.EC2API) (bool, error)
+}
+
+var _ NetworkManager = (*NetworkProvider)(nil)
+
+type NetworkProvider struct {
+	logger *logrus.Entry
+}
+
+func NewNetworkManager(logger *logrus.Entry) *NetworkProvider {
+	return &NetworkProvider{
+		logger: logger.WithField("provider", "standalone_network_provider"),
+	}
+}
+
+func (n *NetworkProvider) CreateNetwork() error {
+	/*
+		todo CreateNetwork is called we need to ensure the following
+		we expect _network to exist with a valid cidr
+		the absence of either _network or valid cidr we should return with an informative message
+		re-reconcile until we have a valid _network strat and valid cidr
+	*/
+	logger := n.logger.WithField("action", "CreateNetwork")
+	logger.Debug("CreateNetwork stub")
+	return errorUtil.New("CreateNetwork stub")
+}
+
+func (n *NetworkProvider) DeleteNetwork() error {
+	logger := n.logger.WithField("action", "DeleteNetwork")
+	logger.Debug("DeleteNetwork stub")
+	return nil
+}
+
+/*
+IsEnabled checks for valid rhmi subnets within the cluster VPC
+a valid rhmi subnet will contain a tag with the `organizationTag` value
+
+when rhmi subnets are present in a cluster vpc it indicates that the vpc configuration
+was created in a cluster with a cluster version <= 4.4.5
+
+when rhmi subnets are absent in a cluster vpc it indicates that the vpc configuration has not been created
+and we should create a new vpc for all resources to be deployed in and we should peer the
+resource vpc and cluster vpc
+*/
```

Comment:
I think the why is important here, for future readers. The details can be included at the top of the file.

```suggestion
/*
IsEnabled returns true when no subnets created by the cloud resource operator exist in the openshift cluster vpc.

subnets created by the cloud resource operator are identified by having a tag with the name `<organizationTag>/clusterID`. By default, `integreatly.org/clusterID`.

this check allows us to maintain backwards compatibility with openshift clusters that used the cloud resource operator before this standalone vpc provider was added. If this function returns false, we should continue using the backwards compatible approach of bundling resources in with the openshift cluster vpc.
*/
```